lib/puppet/settings/file_setting.rb

```diff
@@ -75,7 +77,7 @@ def group=(value)
              when "service"
                # Group falls back to `nil` because we cannot assume that a "root" group exists.
                # Some systems have root group, others have wheel, others have something else.
-               Service.new(:group, nil, @settings, :service_group_available?)
+               Service.new(:group, service_group_fallback, @settings, :service_group_available?)
```

Comment:
When reworking the SID code, I found this was a bunch more complicated ... and some of our SID code was quite buggy (and in some cases worked for the wrong reasons).

See #4560 / #4579 / #4603 (PUP-5538), #4564 (PUP-5684),